### PR TITLE
Seed file

### PR DESCRIPTION
**Pull request by DavidBronfen**
#45

### CODE REVIEW

**Comment by DavidBronfen** [comment]
@nirgn975 Ready for review.

**Comment by DavidBronfen** [comment]
@nirgn975 Getting this error and can't find the reason.
Thought maybe it is the categories model that doesn't have the id property in it but I'm still getting the same error after the id have been added.

hyper

Any suggestion here?

**Comment by DavidBronfen** [comment]
The error above fixed by changing the category model path from `../api/Category/categoryModel` to `../api/category/categoryModel`.

**Comment by DavidBronfen** [comment]
@nirgn975 Status:
I'm getting all the categories:
localhost_3000_api_categories

So it works and looks good.

Just one question here about this warning thrown by mongoose:
hyper

Do we need to find any library to use instead of the mongoose built-in promise function?

File: server/index.js [hunk 1]
```diff
@@ -3,4 +3,4 @@ var app = require('./server');
 var logger = require('./util/logger');
 
 app.listen(config.port);
-logger.log('listening on http://localhost:' + config.port);
+logger.log(['listening on http://localhost:' + config.port]);
```
Comment (nirgn975) [inline server/index.js]: Change it to backticks, ES2015 style (without the `+`).

File: server/util/dummyCategories.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  { name_he:'דגים ופירות ים', name_en:'fish and seafood', image_path:'data/categories/fish.png'},
+  { name_he:'מרקים', name_en:'soup', image_path:'data/categories/soup.png'},
+  { name_he:'בשר', name_en:'meat', image_path:'data/categories/meat.png'},
+  { name_he:'עוף', name_en:'chicken', image_path:'data/categories/chicken.png'},
+  { name_he:'צמחוני', name_en:'vegetarian', image_path:'data/categories/vegetarian.png'},
+  { name_he:'טבעוני', name_en:'vegan', image_path:'data/categories/vegan.png'},
+  { name_he:'בריאות ודיאטה', name_en:'health and diet', image_path:'data/categories/health.png'},
+  { name_he:'לחמים', name_en:'bread', image_path:'data/categories/bread.png'},
+  { name_he:'עוגות', name_en:'cakes', image_path:'data/categories/cake.png'},
+  { name_he:'קינוחים', name_en:'desserts', image_path:'data/categories/dessert.png'}
```
Comment (nirgn975) [inline server/util/dummyCategories.js]: Add a comma at the end.

File: server/util/seed.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+  return Promise.all(newCategories)
+  .then((savedCategories) => {
+    return _.merge({ categories: savedCategories }, data || {});
+  });
+};
+
+cleanDB()
+.then(createCategories)
```
Comment (nirgn975) [inline server/util/seed.js]: indent.

File: server/util/seed.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+.then(logger.log.bind([logger]))
+.catch(logger.log.bind([logger]));
```
Comment (nirgn975) [inline server/util/seed.js]: indent.

File: server/util/seed.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  return Promise.all(newCategories)
+  .then((savedCategories) => {
+    return _.merge({ categories: savedCategories }, data || {});
+  });
+};
+
+cleanDB()
+.then(createCategories)
+.then(logger.log.bind([logger]))
```
Comment (nirgn975) [inline server/util/seed.js]: indent.

File: server/util/seed.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+const categories = dummyCategories.categories;
+
+logger.log(['Seeding the Database']);
+
+
+// Clean DB a help method to clean the database before each build.
+const cleanDB = () => {
+  logger.log(['... cleaning the DB']);
+  const cleanPromises = [Category]
+  .map((model) => {
```
Comment (nirgn975) [inline server/util/seed.js]: indent.

File: server/util/seed.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+const categories = dummyCategories.categories;
+
+logger.log(['Seeding the Database']);
+
+
+// Clean DB a help method to clean the database before each build.
+const cleanDB = () => {
+  logger.log(['... cleaning the DB']);
+  const cleanPromises = [Category]
```
Comment (nirgn975) [inline server/util/seed.js]: Semicolon at the end.

**Comment by nirgn975** [comment]
@DavidBronfen Are you sure it create the DB and you can query the app? I don't create the DB for me. Maybe we're on different Node.js versions..

About the `DeprecationWarning` try to update [mongoose](https://github.com/Automattic/mongoose/releases) and if it's not working, see this comment: [link](https://github.com/Automattic/mongoose/issues/4951#issuecomment-283327958]

File: server/util/seed.js (hunk 7 above)
Comment (DavidBronfen) [reply server/util/seed.js]: This is not the end, below you can find the `.map((model) => {` that belongs to the same line.

File: server/util/seed.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    .map((model) => {
+      return model.remove().exec();
+    });
+  return Promise.all(cleanPromises);
```
Comment (DavidBronfen) [inline server/util/seed.js]: > Are you sure it create the DB and you can query the app? 

This fix should solve the problem. it was set to `this. cleanPromises `

**Comment by DavidBronfen** [comment]
I tried all the solutions in the ling you've added unfortunately no one worked.

**Comment by DavidBronfen** [comment]
@nirgn975 So here the status is pretty the same, I fixed all the comments you gave me (besides the one I commented back...) and everything should work now.

The warning is still with us I didn't found any solution for that thing.

**Comment by nirgn975** [comment]
@DavidBronfen 
I think, let's ignore the warning for now, let's move on and don't let this delay us. We'll go over this seed issue later on and check what's the standard and how others do it.
I say, let's merge it for know and open a new issue for this to do later on.. What do you say?

**Comment by DavidBronfen** [comment]
@nirgn975 Yalla I'm in if you approve the PR I'll merge it and will concentrate on the test PR.

**Pull request merged**